### ISSUE TITLE
corepack install verbose error

### ISSUE THREAD

**Issue by eichiflSICKAG** [nodejs/corepack#475]
corepack version 0.25.2
node version 20.12.2

I try to install yarn via corepack. This fails with the following error:

```
Adding yarn@4.1.0 to the cache...
Internal Error: Error when performing the request to https://repo.yarnpkg.com/4.1.0/packages/yarnpkg-cli/bin/yarn.js; for troubleshooting help, see https://github.com/nodejs/corepack#troubleshooting
    at fetch (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:22882:11)
    at async fetchUrlStream (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:22914:20)
    at async installVersion (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:23084:18)
    at async Engine.ensurePackageManager (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:23380:32)
    at async InstallLocalCommand.execute (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:23903:5)
    at async InstallLocalCommand.validateAndExecute (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:21173:22)
    at async _Cli.run (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:22148:18)
    at async Object.runMain (/home/user/.nvm/versions/node/v20.12.2/lib/node_modules/corepack/dist/lib/corepack.cjs:24266:12)
```

troubleshooting is all done.
- curling this url (`https://repo.yarnpkg.com/4.1.0/packages/yarnpkg-cli/bin/yarn.js`) works
- NO_PROXY is set to `*yarnpkg.com*`

I have no idea how to debug this further, as no `verbose` flag is available and corepack is super silent about the reasons if it's failing.

**Comment by aduh95** [comment]
Can you try installing the latest version of Corepack? I've encountered this error before, and send a fix – although it was related to a proxy being present, so maybe yours is different, but anyway, trying with the latest version would help with debugging

**Comment by eichiflSICKAG** [comment]
@aduh95 yes this works perfectly with corepack 28. Also works with 22 but not 25 shipped with node 20.12.2

**Comment by aduh95** [comment]
Good to hear! There's a 20.13.0 version of Node.js in preparation that includes Corepack 0.28, this should be fixed soon. (Node.js 20.13.0 is scheduled for release on May 7th, although dates are subject to change). Closing for now, thanks for the report.
